fabric-rendering-v1/src/testmodClient/java/net/fabricmc/fabric/test/rendering/client/HudRenderEventsTests.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.test.rendering.client;
+
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.util.Colors;
+import net.minecraft.util.Identifier;
+
+import net.fabricmc.api.ClientModInitializer;
+import net.fabricmc.fabric.api.client.rendering.v1.HudLayerRegistrationCallback;
+import net.fabricmc.fabric.api.client.rendering.v1.IdentifiedLayer;
+
+public class HudRenderEventsTests implements ClientModInitializer {
+	@Override
+	public void onInitializeClient() {
+		HudLayerRegistrationCallback.EVENT.register(layeredDrawer -> layeredDrawer.addLayerBefore(IdentifiedLayer.MISC_OVERLAYS, Identifier.of("fabric", "test_before_misc_overlay"), (context, tickCounter) -> {
```

Comment:
This is quite hard to read, I would split the render code out into their own functions in the same way that vanilla does.